src/main/java/io/supertokens/jwt/JWTSigningFunctions.java

```diff
@@ -83,8 +83,8 @@ public static String createJWTToken(Main main, String algorithm, JsonObject payl
         headerClaims.put("typ", "JWT");
         headerClaims.put("kid", keyToUse.keyId);
 
-        long currentTimeInMillis = System.currentTimeMillis();
-        long jwtExpiry = (currentTimeInMillis / 1000) + (jwtValidity); // JWT Expiry is seconds from epoch not millis
+        double currentTimeInMillis = System.currentTimeMillis();
+        double jwtExpiry = (currentTimeInMillis / 1000) + (jwtValidity); // JWT Expiry is seconds from epoch not millis
```

Comment:
```
long currentTimeInMillis = System.currentTimeMillis();
long jwtExpiry = roundUp((currentTimeInMillis / 1000.0) + (jwtValidity)); // JWT Expiry is seconds from epoch not millis
```

Reply:
Also need to add test for this - if you can

Reply:
Changed the logic and all existing tests pass